### ISSUE TITLE
Clicking right side of inserted CKEditor5 element results in extra empty element

### ISSUE THREAD

**Issue by pmagunia** [ckeditor/ckeditor5#13507]
## 📝 Provide detailed reproduction steps (if any)

I asked this question on StackOverflow but did not get an answer so re-asking here. 

https://stackoverflow.com/questions/75502754/clicking-right-side-of-inserted-ckeditor5-element-results-in-extra-empty-element

I'm trying to insert a button in CKEditor5. Most of it is working OK but when I click on the right side of an inserted element it results in an extra empty element.

Here is the YouTube video of the behavior: https://www.youtube.com/watch?v=XXLyJcJJk00

...And here is the code to insert the View Fragment:

```
editor.model.change(writer => {
  const viewFragment = editor.data.processor.toView('<a href="' + value.url + '" class="btn btn-' + value.type.toLowerCase() + ' simple-button-plugin">' + value.text + '</a>');
  const modelFragment = editor.data.toModel(viewFragment);
  writer.model.insertContent(modelFragment);
  writer.setSelection( editor.model.document.getRoot(), 'end' );
});
```
### ✔️ Expected result

The extra empty element is not added.

### ❌ Actual result

You can see the result here. There are a number of `&NoBreak;`

![](https://i.stack.imgur.com/qeu1J.png)

### ❓ Possible solution

I'm not sure about possible solutions

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by LavrovE** [comment]
any updates?

**Comment by Witoso** [comment]
Hi! This is a standard behavior of the editor in the editing mode. Such placeholders allow moving selection (caret) to a proper place. When you get the HTML data, this will be removed.

**Comment by Witoso** [comment]
Cannot reproduce in our demos, looks this was solved in #15051 (`v40.2`).

**Comment by Witoso** [comment]
To add on top of this, `&NoBreak`s are added only in editing to allow selection movement, it will not be added to the HTML data.
